src/win/tty.c

```diff
@@ -1878,6 +1878,27 @@ int uv_tty_write(uv_loop_t* loop,
 }
 
 
+int uv__tty_try_write(uv_tty_t* handle,
+                      const uv_buf_t bufs[],
+                      unsigned int nbufs) {
+  DWORD error;
+  unsigned int i;
+  int bytes;
+
+  if (handle->write_reqs_pending > 0)
+    return UV_EAGAIN;
+
+  if (uv_tty_write_bufs(handle, bufs, nbufs, &error))
+    return uv_translate_sys_error(error);
+
+  bytes = 0;
+  for (i = 0; i < nbufs; i++)
+    bytes += bufs[i].len;
+
+  return bytes;
```

Comment:
That's possible although uv__count_bufs is currently unix-only. I can move it to uv-common.